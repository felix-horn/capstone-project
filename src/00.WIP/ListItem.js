import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import Checkbox from '@material-ui/core/Checkbox'
import { InputBase } from '@material-ui/core'

ListItem.propTypes = {
  value: PropTypes.string.isRequired,
  handleCheckboxClick: PropTypes.func.isRequired,
  isChecked: PropTypes.bool,
}

function ListItem({ value, handleCheckboxClick, isChecked }) {
  return (
    <StyledListItem>
      <Checkbox
        color="default"
        checked={isChecked}
        onChange={handleCheckboxClick}
      />

      <InputBase name={value} value={value} />
    </StyledListItem>
  )
}

const StyledListItem = styled.div`
  display: flex;
  align-items: center;
`

export default ListItem
