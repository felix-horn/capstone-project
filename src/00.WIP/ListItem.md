```js padded
<ListItem isChecked=""/>
<ListItem isChecked="" value="Butter"/>
<ListItem isChecked="true" value="Milk"/>
```