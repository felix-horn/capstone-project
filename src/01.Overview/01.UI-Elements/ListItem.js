import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
}

export default function ListItem({ title }) {
  return (
    <ListItemLayout>
      <Checkbox />
      <Title>{title}</Title>
    </ListItemLayout>
  )
}

const ListItemLayout = styled.label`
  display: flex;
  align-items: center;
  gap: 5px;
`

const Checkbox = styled.div`
  border: 1px solid var(--light-gray);
  border-radius: 2px;
  height: 15px;
  width: 15px;
`

const Title = styled.p`
  margin: 0;
  color: var(--dark-gray);
`
