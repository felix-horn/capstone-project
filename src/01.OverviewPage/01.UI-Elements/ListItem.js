import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
}

export default function ListItem({ title }) {
  return (
    <ListItemStyled>
      <CheckboxStyled />
      <TitleStyled>{title}</TitleStyled>
    </ListItemStyled>
  )
}

const ListItemStyled = styled.label`
  display: flex;
  align-items: center;
  gap: 5px;
`

const CheckboxStyled = styled.div`
  border: 1px solid var(--light-gray);
  border-radius: 2px;
  height: 15px;
  width: 15px;
`

const TitleStyled = styled.p`
  margin: 0;
  color: var(--dark-gray);
`
