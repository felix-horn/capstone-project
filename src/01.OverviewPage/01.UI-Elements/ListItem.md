```js padded
<ListItem title="Pommes frites"/>
<ListItem title="Ketchup"/>
```