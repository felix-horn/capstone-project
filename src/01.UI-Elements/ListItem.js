import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import Checkbox from '@material-ui/core/Checkbox'
import CloseIcon from '@material-ui/icons/Close'
import { useState } from 'react'

ListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
  onToggle: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  onEnter: PropTypes.func.isRequired,
}

export default function ListItem({
  id,
  title,
  isChecked,
  handleChange,
  onToggle,
  onDelete,
  onEnter,
}) {
  const [isDeleteIconShown, setIsDeleteIconShown] = useState(false)

  return (
    <ListItemStyled checked={isChecked} data-testid="list-item">
      <Checkbox
        type="checkbox"
        color="default"
        checked={isChecked}
        onChange={onToggle}
        data-testid="checkbox"
      />
      <TitleStyled
        name={id}
        value={title}
        onChange={handleChange}
        onKeyUp={(e) => e.key === 'Enter' && handleEnter(e)}
        onFocus={() => setIsDeleteIconShown(true)}
        onBlur={() => setTimeout(() => setIsDeleteIconShown(false), 1)}
        autoFocus={true}
        data-testid="title-list-item"
      />
      {isDeleteIconShown && (
        <DeleteButtonStyled onClick={onDelete} data-testid="delete-list-item" />
      )}
    </ListItemStyled>
  )

  function handleEnter(event) {
    event.target.blur()
    onEnter()
  }
}

const ListItemStyled = styled.label`
  width: 80vw;
  display: flex;
  align-items: center;
  text-decoration: ${(props) => (props.checked ? 'line-through' : 'none')};
  opacity: ${(props) => (props.checked ? 0.5 : 1)};
`
const TitleStyled = styled.input`
  margin-left: 9px;
  border: none;
  &:focus {
    outline: none;
  }
`
const DeleteButtonStyled = styled(CloseIcon)`
  margin-left: auto;
  transform: scale(0.9);
  opacity: 0.4;
`
