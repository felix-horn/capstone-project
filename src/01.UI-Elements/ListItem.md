```js padded
<ListItem titleListItem="Milk"/>
<ListItem titleListItem="Butter" checked />

```