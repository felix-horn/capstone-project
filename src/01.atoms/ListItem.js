import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  titleListItem: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
  onCheckboxClick: PropTypes.func.isRequired,
}
function ListItem({ titleListItem, isChecked, onCheckboxClick }) {
  return (
    <LabelStyled>
      <Checkbox
        type="checkbox"
        checked={isChecked}
        onChange={onCheckboxClick}
      />
      <span>{titleListItem}</span>
    </LabelStyled>
  )
}

const LabelStyled = styled.label`
  display: block;
`

const Checkbox = styled.input`
  font-size: 1em;
`

/** @component */
export default ListItem
