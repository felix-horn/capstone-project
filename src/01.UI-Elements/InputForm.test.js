import { render } from '@testing-library/react'

import ListItem from './ListItem'

describe('ListItem', () => {
  it('shows the right item title', () => {
    const { container } = render(<ListItem titleListItem="Milk" checked />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
