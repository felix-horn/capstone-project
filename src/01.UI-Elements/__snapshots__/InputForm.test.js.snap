// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem shows the right item title 1`] = `
<div
  class="ListItem__Wrapper-xw4fk9-1 hGfRwN"
>
  <label>
    <input
      class="ListItem__BigCheckbox-xw4fk9-0 CeCUz"
      type="checkbox"
    />
    <span>
      Milk
    </span>
    <button
      class="ListItem__TrashButton-xw4fk9-2 fmmKOx"
    />
  </label>
</div>
`;
